script.js

```diff
@@ -1,12 +1,155 @@
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  episodeList.map((el) => {
+    let episodeDivContainer = document.createElement("div");
+    let episodeTitleElement = document.createElement("h3");
+    let episodeImageElement = document.createElement("img");
+    let episodeSummary = document.createElement("span");
+
+    episodeDivContainer.classList.add("divClass");
+
+    if (el.season < 10) {
+      el.season = `0${el.season}`;
+    }
+    if (el.number < 10) {
+      el.number = `0${el.number}`;
+    }
+
+    episodeTitleElement.innerHTML = `${el.name} - S${el.season}E${el.number}`;
+    episodeImageElement.src = `${el.image.medium}`;
+    episodeSummary.innerHTML = `${el.summary}`;
+
+    document.getElementById("root").appendChild(episodeDivContainer);
+    episodeDivContainer.appendChild(episodeTitleElement);
+    episodeDivContainer.appendChild(episodeImageElement);
+    episodeDivContainer.appendChild(episodeSummary);
+  });
 }
 
-window.onload = setup;
+function searchShow(query) {
+  const url = `https://api.tvmaze.com/search/shows?q=${query}`;
+  fetch(url)
+    .then((response) => {
+      if (response.status >= 200 && response.status < 400) {
+        return response.json();
+      } else {
+        throw "HTTP ERROR";
+      }
+    })
+    .then((jasonData) => {
+      console.log(jasonData);
+      const results = jasonData.map((element) => ({
+        name: element.show.name,
+        image: element.show.image.medium,
+        summary: element.show.summary,
+      }));
+      renderResults(results);
+      document.getElementById("errorMessage").innerHTML = "";
+    })
+    .catch((error) => {
+      document.getElementById("errorMessage").innerHTML = error;
+      renderResults([]);
+    });
+}
+
+// show the dropdown list of episodes
+function ListEpisode(allEpisodes) {
+  allEpisodes.map((episode) => {
+    let optionElement = document.createElement("option");
+    optionElement.innerHTML = `${episode.name} - S${episode.season}E${episode.number}`;
+    selectElement.appendChild(optionElement);
+  });
+}
+
+//This function will show the dorpdown list of all shows
+function ListOfShows(allShows) {
+  showList = document.getElementById("selectShow");
+  let selectElement = document.createElement("select");
+
+  selectElement.addEventListener("change", (event) => {
+    selectedShowId = event.target.value;
+    console.log(selectedShowId);
+
+    const theResultOfEpisode = // await showtheEpisodesList(selectedShowId);
+    console.log(theResultOfEpisode);
+    ListEpisode(theResultOfEpisode);
+  });
+
+  allShows.map((show) => {
+    let optionElement = document.createElement("option");
+    optionElement.setAttribute("value", show.id);
+    optionElement.innerText = `${show.name}`;
+    selectElement.appendChild(optionElement);
+  });
+  showList.appendChild(selectElement);
+}
+
+// this function is using fetch to return the eposide list
+function showtheEpisodesList(selectedShowId) {
```

Comment:
the function does not return any values currently. Consider using async await rather than promise.then, as its much easier to return values using the await syntax

Reply:
these values can then also be used to update the page with just the episodes for that show